src/zenml/orchestrators/input_utils.py

```diff
@@ -64,10 +65,13 @@ def resolve_step_inputs(
             )
 
         try:
-            artifact = step_run.outputs[input_.output_name]
+            os_ = step_run.outputs[input_.output_name]
+            artifact = [
+                o for o in os_ if o.save_type == ArtifactSaveType.STEP_OUTPUT
+            ].pop()
```

Comment:
This will not raise a `KeyError` but an index error. I think we should also check that `len(list) < 2` just to be sure, in case we ever allow returning two regular artifacts with the same name, to avoid any ambiguity here.